joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -36,9 +36,11 @@ trait AstCreatorHelper { this: AstCreator =>
         parserNodeCache.addOne(json(ParserKeys.NodeId).num.toLong, pni)
         node match
           case CallExpr =>
-            json(ParserKeys.Fun)(ParserKeys.Obj).objOpt.map(obj => {
-              createParserNodeInfo(obj(ParserKeys.Decl))
-            })
+            Try(json(ParserKeys.Fun)(ParserKeys.Obj)(ParserKeys.Decl)) match
+              case Success(obj) =>
+                // This is being called only to cache this objects
+                createParserNodeInfo(obj)
```

Comment:
Instead of a match statement with an empty default case, you could use a `.foreach` maybe?

Reply:
Sometimes there is no `ParserKeys.Obj` key some other times there is no `ParserKeys.Decl`. This Try takes care of multiple combinations, which might throw a null pointer exception.